src/analyze.jl

```diff
@@ -157,10 +159,91 @@ function _slic(alg::SLIC, img::AbstractArray{<:Lab, 2})
         segments ./= n_segment_elems # broadcast: (n_segments,) -> (n_segments, 5)
     end
 
+    # Enforce connectivity
+    # Reference: https://github.com/scikit-image/scikit-image/blob/7e4840bd9439d1dfb6beaf549998452c99f97fdd/skimage/segmentation/_slic.pyx#L240-L348
     if alg.enforce_connectivity
-        nothing # TODO
+        println("zergling")
+        segment_size = height * width / n_segments
+        min_size = round(Int, 0.5 * segment_size)
+        max_size = round(Int, 3.0 * segment_size)
+
+        dy = [1, -1, 0, 0]
+        dx = [0, 0, 1, -1]
+        # dz = [] # reversed for supervoxels
+
+        start_label = 1
+        mask_label = start_label - 1 # indicates the label of this pixel has not been assigned
+        nearest_segments_final = fill(mask_label, height, width)
+        current_new_label = start_label
+
+        # used for BFS
+        current_segment_size = 1
+        bfs_visited = 0
+
+        # store neighboring pixels
+        # now set the dimension to 2 because we are using superpixel
+        coord_list = fill(0, max_size, 2)
+ 
+        for x = 1:width
+            for y = 1:height
+                if nearest_segments[y, x] == mask_label continue end
+                if nearest_segments_final[y, x] > mask_label continue end
+
+                adjacent = 0
+                label = nearest_segments[y, x]
+                nearest_segments_final[y, x] = current_new_label
+                current_segment_size = 1
+                bfs_visited = 0
+                coord_list[bfs_visited + 1, 1] = y
+                coord_list[bfs_visited + 1, 2] = x
+
+                # Preform BFS to find the size of superpixel with 
+                # same lable number
+                while bfs_visited < current_segment_size <= max_size
```

Comment:
It might be worth adding `@inbounds` to this deeply nested for loop so as to skip unnecessary checks during `getindex` and `setindex`, e.g., `yy = coord_list[bfs_visited + 1, 1] + dy[i]` and `coord_list[current_segment_size + 1, 1] = yy`

```suggestion
                @inbounds while bfs_visited < current_segment_size <= max_size
```

Reply:
Though it sounds great, BFS usually needs to check bound.
I will try to research about BFS non-checking bounds method.